contrib_bots/lib/github.py

```diff
@@ -1,25 +1,56 @@
+#!/usr/bin/env python
+
 # The purpose of this file is to handle all requests
 # to the Github API, which will make sure that all
 # requests are to the same account, and that all requests
 # authenticated correctly and securely
 # The sole purpose of this is to authenticate, and it will
 # return the requests session that is properly authenticated
+import logging
 import os
 import requests
 import six.moves.configparser
 
-CONFIG_FILE = '~/.github-issue-bot/github-issue-bot.conf'
+
+# This file contains the oauth token for a github user, their username, and optionally, a repository
+# name and owner. All in the format:
+# github_repo
+# github_repo_owner
+# github_username
+# github_token
+CONFIG_FILE = '~/.github-auth.conf'
+
+def __init__():
+    global config
+    config = six.moves.configparser.ConfigParser()  # Sets up the configParser to read the .conf file
+    config.read([os.path.expanduser(CONFIG_FILE)])  # Reads the config file
+
 
 def auth():
-    # Sets up the configParser to read the .conf file
-    config = six.moves.configparser.ConfigParser()
-    # Reads the config file
-    config.read([os.path.expanduser(CONFIG_FILE)])
-    # Gets the properties from the config file
-    oauth_token = config.get('github', 'github_token')
-    username = config.get('github', 'github_username')
     # Creates and authorises a requests session
     session = requests.session()
-    session.auth = (username, oauth_token)
-    # Returns the session
+    session.auth = (get_username(), get_oauth_token())
+
     return session
+
+
+def get_oauth_token():
+    return config.get('github', 'github_token')
```

Comment:
Probably want to do exception handling here as well. I think you need to check what happens when either 'github' or 'github_token' does not exist in the config and then do exception handling if it is needed.